### PR TITLE
feat: add loading state to useWeb3ModalState hook

### PR DESCRIPTION
**Pull request by enesozturk**
# Breaking Changes

We are storing the `loading` state inside our scaffold which is managed in `PublicStateController` to make conditional renderings in UI depending on the client's (ethers/wagmi) loading statuses. While `open`/`close` states are exposed to hooks for developers, this `loading` state not exposed. 

This PR introduces changes to share the `loading` state to the hooks.

# Changes

- feat: add `loading` state to `useWeb3ModalState` hook

### CODE REVIEW

**Review by tomiir — APPROVED** [review]
lgtm!

**Comment by tomiir** [comment]
@boidushya we'd need some docs update on this right?

**Comment by boidushya** [comment]
> @boidushya we'd need some docs update on this right?

yep! This relevant to React, Next & JS right? I'll go ahead and open a PR.

**Comment by enesozturk** [comment]
@tomiir @svenvoskamp @ignaciosantise I think not it's handling loading better. Could you review it again?

**Review by ignaciosantise — APPROVED** [review]
🚀 looking good!

just left you a question

**Comment by glitch-txs** [comment]
Can we add a small description on what this state is tracking? It's not 100% clear to me

File: packages/scaffold/src/client.ts [hunk 1]
```diff
@@ -130,7 +130,7 @@ export class Web3ModalScaffold {
   }
 
   public getState() {
-    return { ...PublicStateController.state }
+    return PublicStateController.state
```
[the remark this answers is not in the document]
Comment (enesozturk) [reply packages/scaffold/src/client.ts:121]: Since we don't expose any other value, we don't have to do destruction here, seems unnecessary.

**Pull request merged**